account_invoice_ubl/models/account_move.py

```diff
@@ -39,10 +39,17 @@ def _ubl_add_header(self, parent_node, ns, version="2.1"):
 
     def _ubl_add_order_reference(self, parent_node, ns, version="2.1"):
         self.ensure_one()
-        if self.name:
-            order_ref = etree.SubElement(parent_node, ns["cac"] + "OrderReference")
+        order_id = self.line_ids.mapped("sale_line_ids.order_id")
+        if len(order_id) > 1:
+            logger.warning("UBL Invoice for multiple sale order")
+            order_id = order_id[0]
```

Comment:
Instead of getting one order only, you could get all the orders and merge all the `client_order_ref`'s together

Reply:
Anyway this is just a technical suggestion, not sure if it's compliant with UBL functional specifications :see_no_evil: